samza-api/src/main/java/org/apache/samza/runtime/ApplicationRunner.java

```diff
@@ -16,34 +16,35 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.samza.system;
+package org.apache.samza.runtime;
 
 import java.lang.reflect.Constructor;
 import org.apache.samza.annotation.InterfaceStability;
 import org.apache.samza.config.ConfigException;
 import org.apache.samza.operators.StreamGraphBuilder;
 import org.apache.samza.config.Config;
+import org.apache.samza.system.StreamSpec;
 
 
 /**
  * Interface to be implemented by physical execution engine to deploy the config and jobs to run the {@link org.apache.samza.operators.StreamGraph}
  *
  * Implementations of this interface must define a constructor with a single {@link Config} as the argument in order
- * to support the {@link ExecutionEnvironment#fromConfig(Config)} static constructor.
+ * to support the {@link ApplicationRunner#fromConfig(Config)} static constructor.
  */
 @InterfaceStability.Unstable
-public interface ExecutionEnvironment {
+public interface ApplicationRunner {
 
-  String ENVIRONMENT_CONFIG = "job.execution.environment.class";
-  String DEFAULT_ENVIRONMENT_CLASS = "org.apache.samza.system.StandaloneExecutionEnvironment";
+  String RUNNER_CONFIG = "job.runner.class";
+  String DEFAULT_RUNNER_CLASS = "org.apache.samza.runtime.AbstractApplicationRunner";
```

Comment:
Default should be a concrete implementation. Maybe default to Yarn or Standalone?